src/tests/unit/doughs-dao.test.js

```diff
@@ -179,4 +184,52 @@ describe('test doughs dao', () => {
       });
     });
   });
+  context('getDoughById', () => {
+    let result;
+    beforeEach(() => {
+      connectionStub = sinon.stub().returns(executeReturn);
+      doughsDao = proxyquireDoughsDao(
+        connectionStub,
+        serializeDoughsStub,
+        serializeDoughStub,
+      );
+    });
+    context('when called with an integer id', () => {
+      beforeEach(async () => {
+        result = doughsDao.getDoughById(1);
+        await result;
+      });
+
+      it('calls execute with the right query and bind params', () => {
+        connectionStub.should.have.been.calledWith(
+          getDoughByIdQuery,
+          { id: 1 },
+        );
+      });
+
+      it('extracts rows from the return and passes them to the serializer', () => {
+        serializeDoughStub.should.have.been.calledWith(
+          executeReturnRows,
+          'doughs/1',
+        );
+      });
+
+      it('returns the result from the serializer', async () => result
+        .should.eventually.deep.equal(baseGetDoughsReturn));
+
+      context('when the database returns an empty result', () => {
+        beforeEach(async () => {
+          connectionStub.returns(emptyDatabaseReturn);
+          await doughsDao.getDoughById(1);
+        });
+
+        it('passes an empty array to the serializer', () => {
+          serializeDoughStub.should.have.been.calledWith(
+            [],
+            'doughs/1',
+          );
+        });
+      });
+    });
+  });
```

Comment:
You probably also wanna assert all number fields have been converted to integer in response. I also notice this branch is not based on the latest change from the last PR, so that the returned response in this PR is not correct (number fields are still strings). Can you merge the last PR to this branch?

Reply:
That's a behavior that happens internally to the serializer, so I didn't think it made sense to test for it in this case. For the purpose of writing basically all of my DAO tests I've assumed that the serializer output is correct and the DAO doesn't touch the output of the serializer.

Reply:
Oh sorry for leaving the comment in the wrong place. I was mainly saying we should also have unit tests for serializer. Are you planning to work on it on the other ticket?

Reply:
There are tests for the serializer but they need to be updated to actually test that they're converting into the string. I can open another ticket for that if you think that's the way to go.

Reply:
Either way works for me. The reason I'm asking it is because the title of this PR sounds like it should cover all testing cases for `GET /doughs/{id}`.

Reply:
That's a good point. I'm fine saying that's in scope for this ticket and doing that work in this PR.